caprover/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,138 @@
+s4qs8pkyx1eh   captain-captain      replicated   1/1        caprover/caprover:1.13.1
+qwmynacoiol5   captain-nginx        replicated   1/1        nginx:1.24
+da7k7crhdlsb   srv-captain--files   replicated   1/1        filebrowser/filebrowser:v2     # <--- 1/1
+```
+
+## CapRover has crashed and you need to access app config
+
+All of the app definitions, including environmental variables, are stored in a file on disk:
+
+```
+$ sudo cat /captain/data/config-captain.json
+```
+
+This could be useful if you need to set up CapRover, and all one-click apps, anew.
+For example, if one of the apps (like Apache Superset) was set up with a secret key used to encrypt sensitive information in the database, and you need to retrieve this key to set up the app again (see [**Lost Superset `SECRET KEY`**](#lost-superset-secret_key))
+
+## Windmill worker apps are not starting
+
+It has been observed that the Windmill worker apps don't start if the DB connection string does not have `?sslmode=require` appended, if connecting to a database that requires SSL. This is not added to the DB connection string by default.
+
+## Superset metastore database not created after installation
+
+It may be necessary to manually create the `superset_metastore` database by running the following as a PostgreSQL admin user:
+
+```sql
+CREATE DATABASE superset_metastore;
+```
+
+## Lost Superset `SECRET_KEY`
+
+Apache Superset requires a `SECRET_KEY` for securing sessions and other cryptographic functions such as encrypting sensitive information in the database.
+
+Our [Superset one-click app](one-click-apps/v4/apps/superset-only.yml) defaults to using a new `SECRET_KEY` on each deploy. If you re-deploy Superset against an existing database with a different `SECRET_KEY`, you will see an error on the front end: "An error occurred while fetching databases: Invalid decryption key,' and in the logs the same ValueError will be raised.
+
+If you need to retrieve a lost `SECRET_KEY` from a previous CapRover deployment (e.g. if CapRover has fatally crashed and you need to re-set up CapRover and Superset), one option is to find it in `config-captain.json` as described in [CapRover has crashed and you need to access app config](#caprover-has-crashed-and-you-need-to-access-app-config).
+
+Barring that, know that very few tables in the Superset metastore use this SECRET_KEY, the
+most critical one being `dbs` (to encrypt its columns "password" and "encrypted_extra").
+And `dbs` stores the Database Connections, which presumably you are able to create from scratch
+via the Superset webapp > Settings > "Database Connections" -- this time with the correct (new) SECRET_KEY.
+If charts and dashboards rely on an old Database Connection that was encrypted with a different SECRET_KEY,
+you could then modify that old entry in the `dbs` table by overwriting its "password" and "encrypted_extra"
+with the values of the new entry you just created.
```

Comment:
This is not a comment about this PR, but just something I've noticed across our repos and is visually obvious here - I wonder if we should enforce line length limits for markdown files.